geode-lucene/src/test/java/org/apache/geode/cache/lucene/internal/distributed/DistributedScoringJUnitTest.java

```diff
@@ -110,15 +109,8 @@ public void uniformDistributionProducesComparableScores() throws Exception {
 
     List<EntryScore<String>> distResult = manager.reduce(collectors).getEntries().getHits();
 
-    Assert.assertEquals(singleResult.size(), distResult.size());
     Assert.assertTrue(singleResult.size() > 0);
-
-    for (Iterator single = distResult.iterator(), dist = singleResult.iterator(); single.hasNext()
```

Comment:
The scoring is now not consistent for single node compared to distributed nodes... there is a way to configure the old scoring mechanism to keep them consistent... if we want to move to the new scoring system, we'll have to figure out a way to feed the correct metadata to the score algorithm...

Reply:
Do we need to change `reduce` method to compare on key instead of score? Or it is enough to see that all entries exist in both singleResult and distResult?

Reply:
I guess it depends on what we consider acceptable.  I would think we'd want to modify the scoring to be consistent whether we have a 1 node cluster or a multinode cluster.  The classic scoring in lucene seemed to work for us in 6.0 but it looks like things have changed with their scoring.

This might be the relevant page. 
 It looks like we'd have to set the similarity in the index to use the older (I think tdidf one?) http://lucene.apache.org/core/8_0_0/core/org/apache/lucene/search/similarities/Similarity.html

Reply:
In case we want to use similarity(TFIDFSimilarity) we need to implement three methods: _tf, idf_ and _lengthNorm_.
There is a good article about tf-idf and BM25(default in lucene): https://opensourceconnections.com/blog/2015/10/16/bm25-the-next-generation-of-lucene-relevation/

I added a new commit with implementation of these methods.

Reply:
In case we want to use similarity(TFIDFSimilarity) we need to implement three methods: _tf, idf_ and _lengthNorm_.
There is a good article about tf-idf and BM25(default in lucene): https://opensourceconnections.com/blog/2015/10/16/bm25-the-next-generation-of-lucene-relevation/

I added a new commit with implementation of these methods.

Reply:
Awesome.  I think TFIDF was the classic scoring.  I think we should move to the newer scoring but I haven't done any research on how to get there.  If you feel ambitious, by all means, go for it :-)

Either way, nice job!

Reply:
Thanks @jhuynh1,
what is a way forward? Is the latest changes OK or we need to change something else?
Any thoughts/suggestions?

Reply:
Sorry, I got around to reading the implementation a bit more.  Is the sloppyFreq and scoringPayload scoring correctly?  Is it ok to send back 0 and 1 (I know it's acceptable but is that going to give a real score or are we just going to say everything is the same score now?)

Reply:
I think Nabarun had a concern about what type of exception to throw etc.  I'll resolve my conversation if you think the scoring works as intended.   Please check with nabarun about his concerns.  Thanks!

As an aside, i think you originally tried going to lucene 8 and we ran into some issues.  Perhaps those issues are actually resolved now (after doing all the intermediate work to get it this far).

Reply:
sloppyFreq and scoringPayload are deprecated in Lucene 7.x. So in case we change it, it doesn't have any effect on results. 
After this PR is merged I will go to upgrade to lucene 8.
Thanks!